jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1599,3 +1608,24 @@ void gen_nondet_init(
     pointer_type_selector,
     update_in_place);
 }
+
+/// Adds a call for the given method to the end of `assignments`.
+/// \param assignments: A code block that the method call will be appended to.
+/// \param instance_expr: The instance to call the method on. This argument is
+///   ignored if the method is static.
+/// \param method_name: The name of the method to be called.
+void java_object_factoryt::gen_method_call(
+  code_blockt &assignments,
+  const exprt &instance_expr,
+  const irep_idt &method_name)
+{
+  if(const auto func = symbol_table.lookup(method_name))
```

Comment:
Actually it will usually be the case that the method doesn't exist, as most classes don't have both a static initialiser and a `cproverNondetInitialize`, which are the values for `method_name` that this is called with. I renamed this function to `gen_method_call_if_present` to clarify this.